libs/s25main/desktops/dskOptions.cpp

```diff
@@ -570,6 +587,17 @@ void dskOptions::Msg_ButtonClick(const unsigned ctrl_id)
 
             SETTINGS.Save();
 
+            // Does the selected backend support GUI scaling?
+            unsigned recommendedGuiScale;
+            VIDEODRIVER.GetGuiScaleRange(nullptr, nullptr, &recommendedGuiScale);
+            if(VIDEODRIVER.GetGuiScale().percent() == 100
+               && (SETTINGS.video.guiScale != 100 || (SETTINGS.video.guiScale == 0 && recommendedGuiScale != 100)))
```

Comment:
Could be clearer. Really, this tries to figure out if the backend *needs* to support GUI scaling to fulfill the user request and warns if it cannot.